### DISCUSSION TITLE
Copilot tries to write code according to comments located many lines before new comment

### DISCUSSION THREAD

**Discussion by snatchysquid** [community/community#7393] [Copilot Conversations]
This is an issue that doesn't bother me that much, but I thought you'd like to know:
If for example I create a new file with a single comment:
```
# print something
```
the copilot would complete the next line to be:
```
print("Hello World!")
```
But if I have a comment that I left for something I'd like to do later, and a few lines later I write the original comment like so:
```
# get user credentials




# print something
```
then the copilot doesn't suggest anything.

**Comment by Urethramancer** [comment]
It's mostly good to great for me, but it gets streaks of being all over the place too. Sometimes it suggests code I hadn't thought of the need for yet, other times it suggests functions which already exist in the same file. And sometimes I start with a blank file like that and write a comment, but only get more comments.
